Fix/video accessibility

elenamarinaki opened the pull request:
### Why

<!-- What changes are being made? What feature/bug is being fixed here? If you are closing an issue, use the keyword 'resolves' to link the issue automatically -->
Resolves #148 

### What

<!-- what have you done, if its a bug, whats your solution? -->
- added button to all videos
- the videos can be accessed through the `tab` and controlled thought the `spacebar`
- the animated cursor disappears on mobile screens, so there's no user confusion

### Checklist

<!-- Have you done all of these things?  -->

<!--
To check an item, place an "x" in the box like so: "- [x] Documentation"
Remove items that are irrelevant to your changes.
-->

- [ ] Ready to be merged

<!-- feel free to add additional comments -->

joshuaellis commented on packages/web/src/client/styles/stitches.config.ts:
```diff
@@ -89,6 +89,7 @@ const { styled, globalCss, getCssText, config, keyframes } =
     },
     media: {
       smallTabletUp: `(min-width: ${Widths.SmallTablet}px)`,
+      tabletDown: `(max-width: ${Widths.Tablet}px)`,
```
please don't add max-width media breakpoints, its preferred to do mobile up, it's confusing when the two are mixed.

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+  height: '40px',
+  position: 'absolute',
+  display: 'flex',
+  borderRadius: '50%',
```
Wouldn't this mean the outline is really far from the span inside?

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+  width: '40px',
+  height: '40px',
+  position: 'absolute',
+  display: 'flex',
```
Does this need to be flex?

elenamarinaki replied:
@joshuaellis I want to center the image `span` inside. So, I was planning this to be `flex`, so I can apply `justifyContent: 'center'` in the children.

joshuaellis replied:
Justify content is for the children I believe not the children.

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+  borderRadius: '50%',
+  border: 'none',
+  '&:hover': {
+    background: '#e5e5e5',
```
Try not to use hexes, we have a theme for a reason.

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+  span: {
+    width: '24px',
+    justifyContent: 'center',
+    alignItems: 'center',
```
This won't do anything because the display isn't flex?

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+    position: 'relative',
+
+    img: {
+      position: 'absolute',
```
Im confused about to styling decisions here, can you elaborate why you felt this to be necessary?

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -134,3 +144,31 @@ const Video = styled('video', {
+  display: 'flex',
+  borderRadius: '50%',
+  border: 'none',
+  '&:hover': {
```
I think we probably should be checking if we can hover before applying styles?

joshuaellis reviewed (changes requested):
Im assuming the desktop & focus handling is gonna come later?

joshuaellis commented:
### TODO

- [ ] add accessible labels to play pause button
- [ ] investigate how to ensure screen readers know the button belongs to the video in question

elenamarinaki commented:
@joshuaellis 

I checked with various readers to see how they associate information between buttons and their environments.
In readers like `speechify` or `read aloud` any elements that are not visually text are completely ignored.
Using the mac `voice over` I have access to the videos and specifically to the type of element as well as its action, for example `button play`. But I do not have access to the name of the video (assuming it has one - it could be the `id`)
![May-16-2022 11-32-49](https://user-images.githubusercontent.com/76445935/168574325-ad6be5ef-5018-4b93-8581-b0ed1b05bcf2.gif)
---
Regardless, I pass the `video src` as an `id` to the button, as seen in the `Elements` tab:
![Screen Shot 2022-05-16 at 11 34 32](https://user-images.githubusercontent.com/76445935/168574655-d2716ebe-d99e-4750-93cd-fdde14aef54c.png)
![Screen Shot 2022-05-16 at 11 34 52](https://user-images.githubusercontent.com/76445935/168574830-e8ba7baa-1569-4002-932a-2d1d9d4498ae.png)

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -114,6 +148,44 @@ export const VideoPlayer = ({
+        onBlur={handleButtonBlur}
+      >
+        <VideoButtonIcon
+          aria-label={isPaused ? 'Play' : 'Pause'}
```
I don't think I would use an aria-label because an SVG icon isn't focussable

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -114,6 +148,44 @@ export const VideoPlayer = ({
+      <VideoButton
+        style={style}
+        onFocus={handleButtonFocus}
+        onBlur={handleButtonBlur}
```
I would consider adding a visibly hidden span in the button that we change the content of depending on state.

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -114,6 +148,44 @@ export const VideoPlayer = ({
         muted
         playsInline
       />
+      <VideoButton
```
Maybe this should be using https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls where the ID belongs to the button etc.

joshuaellis reviewed (changes requested):
I would probably aim the accessibility enhancements at the button not the span.

joshuaellis commented on packages/web/src/client/components/Video/VideoPlayer.tsx:
```diff
@@ -114,6 +148,45 @@ export const VideoPlayer = ({
+        id={src}
+        aria-label={isPaused ? 'Play' : 'Pause'}
+      >
+        <span hidden>{isPaused ? 'Play' : 'Pause'}</span>
```
```suggestion
        <span css={{ visuallyHidden: '' }}>{isPaused ? 'Play' : 'Pause'}</span>
```

joshuaellis reviewed (changes requested):
see suggestions to fix up the PR :)

joshuaellis reviewed (changes requested):
the background color of the button has disappeared, don't forget to check your branch after you update it 😄

elenamarinaki commented:
@joshuaellis in the `VideoButtonIcon` styled element the `white` was used as a variable, which I am not sure if it was correct, so I changed it. Also I added an extra   `borderRadius: '$circle'` in the `VideoButton` styled element, as when I was loading the page, it had a weird effect of a rectangle transforming into a circle around the button. Now, it starts with a circle, so there's no such issue.
